src/addr.rs

```diff
@@ -89,13 +89,13 @@ impl VirtAddr {
 
     /// Creates a virtual address from the given pointer
     pub fn from_ptr<T>(ptr: *const T) -> Self {
-        Self::new(cast::u64(ptr as usize))
+        Self::new(ptr as u64)
     }
```

Comment:
This function does not have a `#[cfg(target_pointer_width = "64")]`. There are no >64-architectures right now, so this cast should never be truncating in practice, but maybe we should add the cfg attribute anyway?

Reply:
Looks like `cfg` was already missing there. Adding it would technically be a breaking change.

It could also get a `#[cfg(any(target_pointer_width = "32", target_pointer_width = "64")]` to make it 'less breaking'. Not sure if anybody would want to use this function on 32-bit though.

Reply:
Adding `#[cfg(any(target_pointer_width = "32", target_pointer_width = "64")]` with a short comment that `target_pointer_width = "32"` is only added for reducing breakage seems like the best option to me.

Reply:
Sure. Done.